mlflow/transformers.py

```diff
@@ -1601,6 +1596,54 @@ def _coerce_exploded_dict_to_single_dict(self, data):
         else:
             return data
 
+    @staticmethod
+    def _strip_input_from_response_in_instruction_pipelines(input_data, output, output_key):
+        """
+        Parse the output from instruction pipelines to conform with other text generator
+        pipeline types and remove line feed characters and other confusing outputs
+        """
+        replacements = {"\n\n": " ", "A:": ""}
```

Comment:
What would happen if we didn't replace these?

Reply:
Leading and trailing whitespace are probably ok to remove.

Reply:
I'm looking at Dolly's traning data. As you can see, some responses start with `A:`. Is this the reason we need to strip (not replace) `A:`?

<img width="892" alt="image" src="https://user-images.githubusercontent.com/17039389/232677258-005b8c48-df5b-4da0-8420-9d7dc95377fd.png">

Reply:
If anyone uses this in batch mode, they're going to have to parse the output. This is the only generation pipeline that I've found that inserts newline characters in the output. It works for demos and interactive mode, but it's jarring when compared to everything else on HuggingFace

Reply:
I restricted this logic to only be accessible if the underlying Pipeline type is logged in the MLmodel is "InstructionTextGenerationPipeline" so that we don't inadvertently modify any other TextGenerationPipeline outputs

Reply:
Also changing this parse logic to only look at the start of the response string after removing the prompt instead of replacing everywhere.